### PR TITLE
feat(JOML): migrate target system

### PR DESCRIPTION
**Pull request by pollend**
this is a migration of the target system since JOMLUtil is not needed any more in this case.

### CODE REVIEW

File: engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java [hunk 1]
```diff
@@ -158,7 +159,10 @@ public boolean updateBindState(Input input,
                 }
                 if (!keyConsumed) {
                     buttonEvent.prepare(id, ButtonState.DOWN, delta);
-                    buttonEvent.setTargetInfo(target, targetBlockPos, hitPosition, hitNormal);
+                    buttonEvent.setTargetInfo(target,
+                        targetBlockPos == null ? null : JomlUtil.from(targetBlockPos),
+                        hitPosition == null ? null : JomlUtil.from(hitPosition),
+                        hitNormal == null ? null : JomlUtil.from(hitNormal));
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java:165]: Would it make sense to have `JomlUtil.from(null)` return `null` instead?

File: engine/src/main/java/org/terasology/input/cameraTarget/CameraTargetSystem.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ private void updateFocalDistance(HitResult hitInfo, float delta) {
         if (hitInfo.isHit()) {
             Vector3f playerToTargetRay = new Vector3f();
             //calculate the distance from the player to the hit point
-            playerToTargetRay.sub(JomlUtil.from(hitInfo.getHitPoint()), localPlayer.getViewPosition());
+            hitInfo.getHitPoint().sub(JomlUtil.from(localPlayer.getViewPosition()), playerToTargetRay);
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/input/cameraTarget/CameraTargetSystem.java:147]: I guess/hope that "rotation" is on purpose?
So TeraMath's `x.sub(y, z)` equals joml's `y.sub(z, x)`?

File: engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java [hunk 3]
```diff
@@ -176,7 +180,10 @@ public boolean updateBindState(Input input,
                 }
                 if (!keyConsumed) {
                     buttonEvent.prepare(id, ButtonState.UP, delta);
-                    buttonEvent.setTargetInfo(target, targetBlockPos, hitPosition, hitNormal);
+                    buttonEvent.setTargetInfo(target,
+                        targetBlockPos == null ? null : JomlUtil.from(targetBlockPos),
+                        hitPosition == null ? null : JomlUtil.from(hitPosition),
+                        hitNormal == null ? null : JomlUtil.from(hitNormal));
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java:186]: see above

File: engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java [hunk 4]
```diff
@@ -205,7 +212,10 @@ public void update(EntityRef[] inputEntities,
                 boolean consumed = triggerOnRepeat(delta, target);
                 if (!consumed) {
                     buttonEvent.prepare(id, ButtonState.REPEAT, delta);
-                    buttonEvent.setTargetInfo(target, targetBlockPos, hitPosition, hitNormal);
+                    buttonEvent.setTargetInfo(target,
+                        targetBlockPos == null ? null : JomlUtil.from(targetBlockPos),
+                        hitPosition == null ? null : JomlUtil.from(hitPosition),
+                        hitNormal == null ? null : JomlUtil.from(hitNormal));
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/input/internal/BindableButtonImpl.java:218]: see above

**Comment by jdrueckert** [comment]
oh and this PR is not up-to-date with current develop

File: engine/src/main/java/org/terasology/input/cameraTarget/CameraTargetSystem.java (hunk 2 above)
Comment (pollend) [reply engine/src/main/java/org/terasology/input/cameraTarget/CameraTargetSystem.java:147]: for termath x.sub(y,z) --> x = y - z where joml it's  x-y = z. z is the dest in this case.

**Comment by jdrueckert** [comment]
@pollend please add null check inside jomlutil function

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**